TestResultSummaryService/routes/compareTestsOutput.js

```diff
@@ -0,0 +1,59 @@
+const JenkinsInfo = require( '../JenkinsInfo' );
+const {removeTimestamp} = require( './utils/removeTimestamp' );
+const {applyDeepSmithMatch} = require( './utils/applyDeepSmithMatch' );
+
+module.exports = async ( req, res ) => {
+    let { url1, buildName1, buildNum1, ...params } = req.query;
+    const jenkinsInfo = new JenkinsInfo();
+
+    if (!url1 || !buildName1 || !buildNum1) {
+        const errorMsg = "Input parameters are not complete!";
+        res.send( { errorMsg } );
+        return;
+    }
+
+    const url2 = params.url2 ? params.url2 : url1;
+    const buildName2 = params.buildName2 ? params.buildName2 : buildName1;
+    const buildNum2 = params.buildNum2 ? params.buildNum2 : buildNum1;
+
+    let consoleOutput1 = "";
+    let consoleOutput2 = "";
+
+    try {
+        consoleOutput1 = await jenkinsInfo.getBuildOutput(url1, buildName1, buildNum1);
+        consoleOutput2 = await jenkinsInfo.getBuildOutput(url2, buildName2, buildNum2);
+    } catch (error) {
+        const errorMsg = "Failed to obtain Jenkins Outputs!";
+        res.send( { errorMsg } );
+        return;
+    }
+
+    if (!consoleOutput1 || !consoleOutput2) {
+        const errorMsg = "Jenkins Output is empty!";
+        res.send( { errorMsg } );
+        return;
+    }
+
+    if (params.removeTimestampFlag === "true") {
+        consoleOutput1 = removeTimestamp(consoleOutput1);
+        consoleOutput2 = removeTimestamp(consoleOutput2);
+        if(consoleOutput1.error || consoleOutput2.error) {
+            const errorMsg = consoleOutput1.error || consoleOutput2.error;
+            res.send( { errorMsg } );
+            return;
+        }
+    }
+
+    if (params.applyDeepSmithMatchFlag === "true") {
+        consoleOutput1 = applyDeepSmithMatch(consoleOutput1);
+        consoleOutput2 = applyDeepSmithMatch(consoleOutput2);
+        if(consoleOutput1.error || consoleOutput2.error) {
+            const errorMsg = consoleOutput1.error || consoleOutput2.error;
+            res.send( { errorMsg } );
+            return;
+        }
+    }
+
+    const result = (consoleOutput1 === consoleOutput2); 
+    res.send( { result } );        
```

Comment:
Instead of doing several try catch, I think it is easier to have one try catch globally. Put all code into a function, say, `compareTestsOutput()`. 
```
module.exports = async ( req, res ) => {
  try {
      await compareTestsOutput(req, res);
   } catch (error) {
       res.send( { error } );
    }
}

async compareTestsOutput ( req, res) {
....
}
```

In this way, we are able to catch any exceptions.
try catch can be removed in applyDeepSmithMatch.js and removeTimestamp.js

Reply:
New commit has changed try-catch globally as suggested.